invenio_records_resources/services/files/service.py

```diff
@@ -204,13 +192,10 @@ def delete_all_files(self, id_, identity):
         file_keys = [fk for fk in record.files]
         results = [record.files.delete(file_key) for file_key in file_keys]
 
-        self.run_components("delete_all_files", id_, identity, record, results)
+        self.run_components_with_uow(
+            "delete_all_files", uow, id_, identity, record, results)
 
-        record.commit()
-        db.session.commit()
-
-        self.run_components(
-            "post_delete_all_files", id_, identity, record, results)
+        uow.register(RecordCommitOp(record))
```

Comment:
Actually, there might be a bug here because it's not indexing the record afterward.